activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,27 @@
+*   Adds support for `if_not_exists` to `add_column` and `if_exists` to `remove_column.
+
+    Applications can set their migrations to ignore exceptions raised when adding a column that already exists or when removing a column that does not exist.
+
+    Example Usage:
+
+    ```
+    class AddColumnTitle < ActiveRecord::Migration[6.1]
+      def change
+        add_column :posts, :title, :string, if_not_exists: true
```

Comment:
Should we do something in the case where title exists, but it happens to be a `text`, then rolling back the existing column's removed and then migrating again, the column would now be a `string`? Though I suppose that's what you want and later migrations is the newest source of truth. Though it could be potentially confusing and happen silently.

Reply:
Good call. Turns out `column_exists?` takes a type. I've updated this to still raise if you pass a different type and added a test for that case.

Reply:
Woot 🙌